backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/ddorder/api/impl/DDOrderLinesAllocator.java

```diff
@@ -342,11 +356,11 @@ public DDOrderLinesAllocator allocateHUProductStorage(@NonNull final IHUProductS
 		// Shall not happen, but ignore it for now.
 		if (ddOrderLinesToAllocate.isEmpty())
 		{
-			new HUException("No DD Order Lines where found for our product"
-					+ "\n @M_Product_ID@: " + Services.get(IProductBL.class).getProductValueAndName(huProductStorage.getProductId())
-					+ "\n HUProductStorage: " + huProductStorage)
-							.throwOrLogWarning(failIfCannotAllocate, logger);
-			return this;
+			// DD_Order_NoLine_for_product
+			throw HUException.ofAD_Message(MSG_DD_Order_NoLine_for_product)
```

Comment:
this looks very weird :)
* weird param names
* i would prefer do declare IProductBL instance as a class field